packages/plugin-web-attribution-browser/src/web-attribution.ts

```diff
@@ -30,19 +40,63 @@ export const webAttributionPlugin: CreateWebAttributionPlugin = function (option
       const isEventInNewSession = isNewSession(config.sessionTimeout, config.lastEventTime);
 
       if (isNewCampaign(currentCampaign, previousCampaign, pluginConfig, isEventInNewSession)) {
+        const campaignEvent = createCampaignEvent(currentCampaign, pluginConfig);
+        const currentSessionId = config.sessionId ?? -1;
+
         if (pluginConfig.resetSessionOnNewCampaign) {
-          amplitude.setSessionId(Date.now());
+          const nextSessionId = Date.now();
+          amplitude.setSessionId(nextSessionId);
+          campaignPerSession[nextSessionId] = campaignEvent;
           config.loggerProvider.log('Created a new session for new campaign.');
+        } else {
+          campaignPerSession[currentSessionId] = campaignEvent;
         }
-        config.loggerProvider.log('Tracking attribution.');
-        const campaignEvent = createCampaignEvent(currentCampaign, pluginConfig);
-        amplitude.track(campaignEvent);
+
```

Comment:
@justin-fiedler lmk what you think, i updated this part to always cache the campaign event by session id, regardless of configuration. it will be sent on the next event (doesn't have to be a session event) that gets through the timeline.